CRM/Mosaico/MosaicoComposer.php

```diff
@@ -18,8 +18,8 @@ public function createTokenProcessorContext(
     \Civi\FlexMailer\Event\ComposeBatchEvent $e
   ) {
     $context = parent::createTokenProcessorContext($e);
-    // Smarty would break Mosaico CSS.
-    $context['smarty'] = FALSE;
+    // Smarty would break Mosaico CSS unless we know what we are doing (eg. handling via extension)
+    $context['smarty'] = $e->context['smarty'] ?? FALSE;
```

Comment:
If one really wants Smarty (per se) plus Mosaico, then I think this should be a different `template_type`.

1. Do you have a link for the (draft?) extension which handles this well? (Maybe that makes it clearer.)

2. Just to make sure I'm interpreting this correctly: The ultimate value of `$context['smarty']` does *not* stem from `CIVICRM_MAIL_SMARTY` (like in `parent::createTokenProcessorContext()`). This patch implies a new/different flag in `ComposerBatchEvent::$context['smarty']` which determines the activation, and the only way to toggle that is to tap into `civi.flexmailer.compose`. So... there's no way existing deployments would inadvertently flip this behavior on? (So, in that way, the change is safe.)

3. In the same way that `*.tpl` and `*.twig` and `*.jsp` are different file formats, this feels like a different format. Data written in one format (e.g. `*.twig` -- or Mosaico with CiviMail templating) is not interchangeable with data written in the other format (e.g. `*.tpl` -- or Mosaico with Smarty templating) even if they look similar (HTML tags).

4. The Smarty/CSS issue is the most painful problem with `CIVICRM_MAIL_SMARTY` or `$context['smarty'] = TRUE`, but IMHO the semantics of that flag are broken by design. Not because Smarty is bad language -- but because the hybrid of CiviMail-Evaluation-then-Smarty-Evaluation is a needlessly janky way to use it (*broken compiler cache; needing to understand the interaction of multiple languages*). Template evaluation should be atomic/singular (not daisy-chained). There's obviously a pressure/inertia to use it anyway (less ΔSLOC; some users familiar with `CIVICRM_MAIL_SMARTY`). Personally, I'm in a mindset of "avoid that flag at all costs!".